client/my-sites/checkout/checkout-thank-you/redesign-v2/products/domain-product.tsx

```diff
@@ -0,0 +1,90 @@
+import { isDomainTransfer } from '@automattic/calypso-products';
+import formatCurrency from '@automattic/format-currency';
+import { Button, ClipboardButton } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { translate } from 'i18n-calypso';
+import { useState } from 'react';
+import ThankYouProduct from 'calypso/components/thank-you-v2/product';
+import { domainManagementList, domainManagementRoot } from 'calypso/my-sites/domains/paths';
+import type { ReceiptPurchase } from 'calypso/state/receipts/types';
+
+type ThankYouDomainProductProps = {
+	purchase?: ReceiptPurchase;
+	domainName?: string;
+	shareSite?: boolean;
+	siteSlug?: string | null;
+	currency?: string;
+};
+
+type DomainTransferSectionProps = {
+	purchase: ReceiptPurchase;
+	currency: string;
+};
+
+const DomainTransferSection = ( { purchase, currency }: DomainTransferSectionProps ) => {
+	const { __ } = useI18n();
+
+	const purchaseLabel = ( priceInteger: number ) => {
+		if ( priceInteger === 0 ) {
+			return __( 'We’ve paid for an extra year' );
+		}
+
+		const priceFormatted = formatCurrency( priceInteger, currency, {
+			stripZeros: true,
+			isSmallestUnit: true,
+		} );
+
+		return translate( '%(priceFormatted)s for one year', { args: { priceFormatted } } );
```

Comment:
That's more a question for @nightnei but what approach should we generally follow when importing that translate function? As far as I know, instead of importing it directly like this we could also import/use `useTranslate()`. I was wondering that since I saw the `useI18n()` hook above.

Reply:
I've seen different ways of importing the `translate()` function throughout the code base, I'm not sure if there's a particular rule. That said, I do feel it's redundant to import `__()` when we already have `translate()`, so I've made the following update to clean that up:

https://github.com/Automattic/wp-calypso/pull/86186/commits/f4984e5c40ce147cf34235dcdc7dcbc5c74e88ad